### PR TITLE
8338528: GenShen: Cleanup shenandoahHeapRegion

### PR DESCRIPTION
**Pull request by earthling-amzn**
* Break up huge methods in shenandoahHeapRegion
* Clarify comments
* Change confusing method names

## Testing

GHA, internal pipelines: dacapo, extremem, specjbb2015, specjvm2008, dilluvian, heapothesys with and without gc stress flags.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (1 review required, with at least 1 [Committer](https://openjdk.org/bylaws#committer))

### Issue
 * [JDK-8338528](https://bugs.openjdk.org/browse/JDK-8338528): GenShen: Cleanup shenandoahHeapRegion (**Task** - P4)

### Reviewers
 * [Kelvin Nilsen](https://openjdk.org/census#kdnilsen) (@kdnilsen - Committer)
 * [Y. Srinivas Ramakrishna](https://openjdk.org/census#ysr) (@ysramakrishna - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/shenandoah.git pull/476/head:pull/476` \
`$ git checkout pull/476`

Update a local copy of the PR: \
`$ git checkout pull/476` \
`$ git pull https://git.openjdk.org/shenandoah.git pull/476/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 476`

View PR using the GUI difftool: \
`$ git pr show -t 476`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/shenandoah/pull/476.diff">https://git.openjdk.org/shenandoah/pull/476.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/shenandoah/pull/476#issuecomment-2299308903)

### CODE REVIEW

File: src/hotspot/share/gc/shenandoah/shenandoahHeapRegion.cpp [hunk 1]
```diff
@@ -495,60 +492,61 @@ bool ShenandoahHeapRegion::oop_coalesce_and_fill(bool cancellable) {
...
-
-  // Find head.
-  ShenandoahHeapRegion* r = humongous_start_region();
-  assert(r->is_humongous_start(), "need humongous head here");
   assert(CardTable::card_size_in_words() * (words / CardTable::card_size_in_words()) == words,
          "slice must be integral number of cards");
+  return words / CardTable::card_size_in_words();
+}
+
+void ShenandoahHeapRegion::oop_iterate_humongous_slice_dirty(OopIterateClosure* blk,
```
Comment (kdnilsen) [inline src/hotspot/share/gc/shenandoah/shenandoahHeapRegion.cpp:512]: We're removing the dirty_only bool argument?  Was that always true for all callers?  Or always true if write table?

Comment (kdnilsen) [reply src/hotspot/share/gc/shenandoah/shenandoahHeapRegion.cpp:512]: I see, I think.  You've introduced a new method to handle the not-dirty case.

**Review by kdnilsen — APPROVED** [review]

File: src/hotspot/share/gc/shenandoah/shenandoahHeapRegion.cpp (hunk 1 above)
Comment (earthling-amzn) [reply src/hotspot/share/gc/shenandoah/shenandoahHeapRegion.cpp:512]: The original method was split it into two. The code path that handles _all_ cards didn't actually even need to look at the card table. The call sites where `dirty_only == false` call this new method. The call sites where `dirty_only == true` call this `oop_iterate_humongous_slice_dirty`.

**Review by ysramakrishna — APPROVED** [review]
LGTM.

Prior to landing the changes, please record in the PR summary the testing that was done.

**Pull request closed without merging**